### DISCUSSION TITLE
Github Actions workspace is empty

### DISCUSSION THREAD

**Discussion by F21** [community/community#26123] [Actions]
<p>According to the documentation for the GITHUB_WORKSPACE environment variable here: <a href="https://help.github.com/en/articles/virtual-environments-for-github-actions#default-environment-variables" rel="nofollow noopener">https://help.github.com/en/articles/virtual-environments-for-github-actions#default-environment-variables</a>, the Github workspace patch should contain a copy of the source code.</p>
<p>However, this appears to not be the case. The directory is empty when I use it as the working directory and run <code>ls</code>.</p>
<p>My workflow looks like this:</p>
<pre><code>name: Publish the website  
  
on:  
 push:  
 branches:  
 - test-site  
  
jobs:  
 publish:  
 name: Publish site  
runs-on: ubuntu-latest  
steps:  
 - name: Build site assets  
working-directory: ${{ github.workspace }}  
run: ls -la  
- name: Publish site to calcite-site repository  
working-directory: site  
run: echo Publishing..  
</code></pre>
<p>Is Github Actions suppose to include a copy of the repository in the working directory or should I use the checkout action to check a copy out?</p>

**Comment by maxheld83** [comment, marked as answer]
<p>Coming from the “old” (HCL-based) GitHub Actions, I ran into this problem, too.</p>
<p>I think you do have to use the checkout action first, though. That solved my problem.</p>
<p>Like so:</p>
<pre><code>steps:
      - uses: actions/checkout@v1
</code></pre>
<p>I verified that without it <code>ls github/workspace</code> is completely empty.</p>
<p>I think they might have just copied the part of the docs from the old version. Will send an email to support.</p>
<p>Ps.: can’t believe these forums don’t support GitHub markdown syntax :( </p>

**Comment by maxheld83** [comment]
<p>wrote this to support:</p>
<p>Hi octocats,</p>
<p>This is about the new, YML-based GitHub actions.</p>
<p>As per the new docs, I understand that I always need to<a href="https://help.github.com/en/articles/configuring-a-workflow#using-the-checkout-action" rel="nofollow noopener">run a checkout action</a> to use my repo content.</p>
<p>Also in the docs, there’s still <a href="https://help.github.com/en/articles/virtual-environments-for-github-actions#environment-variables" rel="nofollow noopener">this part</a>:</p>
<p>&gt; The GitHub workspace directory path. The workspace directory contains a subdirectory with a copy of your repository.</p>
<p>I think this might be in error, perhaps copy/pasted from the old HCL-based GitHub actions.</p>
<p>If I understand correctly, the new GitHub actions always creates <code>GITHUB_WORKSPACE</code>, maps it, but leaves it empty by default.</p>
<p>Right?</p>

**Comment by F21** [comment]
<p>Thanks for confirming <a class="mention" href="/maxheld83">@maxheld83</a>! It looks like the documentation just needs to be updated.</p>
